tests/kernel/queue/src/test_queue_contexts.c

```diff
@@ -398,3 +408,39 @@ void test_multiple_queues(void)
 		tqueue_get(&queues[i]);
 	}
 }
+
+void user_access_queue_private_data(void *p1, void *p2, void *p3)
+{
+	/* try to access to private kernel data, will happen kernel oops */
+	k_queue_is_empty(&queue);
+
+	ret = TC_FAIL;
```

Comment:
Maybe I am wrong. Is "ret = TC_FAIL" here really necessary? I mean, I didn't see a zassert marco to check the result in the code.   
How about put a ztest_test_pass() inside the k_sys_fatal_handler(), and never return?